src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneArray = [];
+  let cloneState = { ...state };
+
+  for (const item of actions) {
+    switch (item.type) {
+      case 'addProperties':
+        for (const key in item.extraData) {
+          cloneState[key] = item.extraData[key];
+        }
+
+        cloneArray.push({ ...cloneState });
+        break;
+
+      case 'removeProperties':
+        for (const key of item.keysToRemove) {
+          delete cloneState[key];
+        }
+
+        cloneArray.push({ ...cloneState });
+        break;
+
+      case 'clear':
+        cloneState = {};
+        cloneArray.push({ ...cloneState });
+        break;
+    }
```

Comment:
```suggestion
  for (const item of actions) {
    switch (item.type) {
      case 'addProperties':
        for (const key in item.extraData) {
          cloneState[key] = item.extraData[key];
        }

        break;

      case 'removeProperties':
        for (const key of item.keysToRemove) {
          delete cloneState[key];
        }
        
        break;

      case 'clear':
        cloneState = {};
        break;
    }
    
    cloneArray.push({ ...cloneState });
```
your repeated this code this 3 time instead of write after switch once

Reply:
```
    cloneArray.push({ ...cloneState });
```